nilearn/interfaces/fmriprep/tests/utils.py

```diff
@@ -4,24 +4,59 @@
 
 import pandas as pd
 
+from nilearn._utils.data_gen import _create_bids_filename
+from nilearn.interfaces.bids._utils import _bids_entities
 from nilearn.interfaces.fmriprep import load_confounds_utils
 
 img_file_patterns = {
     "ica_aroma":
-        "_space-MNI152NLin2009cAsym_desc-smoothAROMAnonaggr_bold.nii.gz",
-    "regular": "_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz",
-    "res": "_space-MNI152NLin2009cAsym_res-2_desc-preproc_bold.nii.gz",
-    "native": "_desc-preproc_bold.nii.gz",
-    "cifti": "_space-fsLR_den-91k_bold.dtseries.nii",
-    "den": "_space-fsLR_den-32k_desc-preproc_bold.nii.gz",
-    "part": (
-        "_part-mag_space-MNI152NLin2009cAsym_"
-        "desc-preproc_bold.nii.gz"
-    ),
+        {"suffix": "bold",
+         "entities": {"space": "MNI152NLin2009cAsym",
+                      "desc": "smoothAROMAnonaggr"},
+         "extension": "nii.gz"},
+    "regular":
+        {"suffix": "bold",
+         "entities": {"space": "MNI152NLin2009cAsym",
+                      "desc": "preproc"},
+         "extension": "nii.gz"},
+    "res":
+        {"suffix": "bold",
+         "entities": {"space": "MNI152NLin2009cAsym",
+                      "res": "2",
+                      "desc": "preproc"},
+         "extension": "nii.gz"},
+    "native":
+        {"suffix": "bold",
+         "entities": {"desc": "preproc"},
+         "extension": "nii.gz"},
+    "cifti":
+        {"suffix": "bold",
+         "entities": {"desc": "preproc",
+                      "space": "fsLR",
+                      "den": "91k"},
+         "extension": "dtseries.nii"},
+    "den":
+        {"suffix": "bold",
+         "entities": {"space": "fsLR",
+                      "den": "32k",
+                      "desc": "preproc"},
+         "extension": "nii.gz"},
+    "part":
+        {"suffix": "bold",
+         "entities": {"part": "mag",
+                      "space": "MNI152NLin2009cAsym",
+                      "desc": "preproc",
+                      },
+         "extension": "nii.gz"},
     "gifti": (
-        "_space-fsaverage5_hemi-L_bold.func.gii",
-        "_space-fsaverage5_hemi-R_bold.func.gii",
-    ),
+        {"suffix": "bold",
+         "entities": {"hemi": "L",
+                      "space": "fsaverage5"},
+         "extension": "func.gii"},
```

Comment:
Is this a valid BIDS extension ?

Reply:
currently in the BIDS extension proposal for functional derivatives, but I think we also have it in there to test things against the type of files found in fmriprep outputs.

https://bids-specification--519.org.readthedocs.build/en/519/derivatives/functional-derivatives.html#functional-derivatives-maps